### PR TITLE
[stable/mongodb] Feature: explicit service name

### PR DESCRIPTION
**Pull request by cspengl**
#### What this PR does / why we need it:

This PR adds the option to define a hard coded service name. This isn't a good practice in general but can be quite useful for special scenarios. (Referring to [Helm Chart Best Practices](https://github.com/helm/helm/blob/master/docs/chart_best_practices/templates.md#resource-naming-in-templates).
In this case it could be useful to have a "release-independent" service name to tell other applications where to connect with the database. The option is disabled per default and everything works like before - just an additional option.

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/mongodb/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: mongodb
-version: 7.2.8
+version: 7.2.9
```
Comment (miguelaeh) [inline stable/mongodb/Chart.yaml]: Should bump the minor version as you are including a new feature.

File: stable/mongodb/templates/svc-headless-rs.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ template "mongodb.fullname" . }}-headless
+  name: {{ template "mongodb.serviceName" . }}-headless
```
Comment (miguelaeh) [inline stable/mongodb/templates/svc-headless-rs.yaml]: Why do you need to change the headless service too?

File: stable/mongodb/templates/_helpers.tpl [hunk 3]
```diff
@@ -225,3 +225,15 @@ but Helm 2.9 and 2.10 does not support it, so we need to implement this if-else
     {{- end -}}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Returns the proper Service name depending if an explicit service name is set
+in the values file. If the name is set not explicitly it will take the "mongodb.fullname"
```
Comment (miguelaeh) [inline stable/mongodb/templates/_helpers.tpl]:
```diff
- If the name is set not explicitly
+ If the name is not explicitly set
```

File: stable/mongodb/values.yaml [hunk 4]
```diff
@@ -116,6 +116,9 @@ clusterDomain: cluster.local
 
 ## Kubernetes service type
 service:
+  ## Specify an explicit service name. This is not recommended in general and therefore
```
Comment (miguelaeh) [inline stable/mongodb/values.yaml]: Remove this part of the comment as it is already explained in the `_helpers.tpl` 
```diff 
- ## Specify an explicit service name. This is not recommended in general and therefore
- ## disabled by default. The service name will be set to the fullname of the chart.
+ ## Specify an explicit service name.
```

**Review by miguelaeh — CHANGES_REQUESTED** [review]
Thank you very much for the PR @cspengl !!!!!!!
Please take a look to my comments.

**Comment by cspengl** [comment]
Thank you very much for your feedback @miguelaeh !
I am completely fine with the changes you suggested. I will change the code to match your suggestions. Regarding the name of the headless service - I just wanted to hold the service names consistent. When you see any advantages or disadvantages for one of the options please let me know.

**Review by miguelaeh — APPROVED** [review]
Thanks for address the changes!!!

**Pull request merged**